src/main/java/jenkins/scm/api/SCMFile.java

```diff
@@ -40,6 +40,16 @@
  * @author Kohsuke Kawaguchi
  */
 public abstract class SCMFile {
+
+    /**
+     * Cache of the last modified timestamp, to allow repeated calls to minimize the number of network round trips.
+     */
+    private Long modified;
```

Comment:
The modified cache may be overkill. The type cache I see as not being so. I have seen far far too many people do things like

```
File f = ...
if (file.exists() && file.isFile()) {
  ...
}
```

I'll remove the `modified` cache, especially as it is proving difficult for the important implementations of `SCMFileSystem` to be able to get the last modified timestamp of a file without a flurry of API calls.